code/datums/beam.dm

```diff
@@ -90,6 +92,10 @@
 	return ..()
 
 /datum/beam/proc/Draw()
+	if(!get_turf && (!isturf(src.target.loc) || !isturf(src.origin.loc)))		//byond is quirky! if we aren't getting turf tell it to just Not(tm)
+		return
+	var/atom/origin = get_turf? get_turf(src.origin) : src.origin
+	var/atom/target = get_turf? get_turf(src.target) : src.target
```

Comment:
This is probably detrimental and confusing, name these two variables in a different, not omonimous way and early return if one or another are null (which shouldn't be).